### ISSUE TITLE
[Documentation] Updating the User Guide

### ISSUE THREAD

**Issue by oualib** [vertica/VerticaPy#1310]
### What is the requested change?

The User Guide needs to be updated.

We need to convert to RST and remove the notebooks from the doc gen:

**Introduction**

 - [x] best-practices (Umar)
 - [x] installation (Umar)
 - [x] vdf (Umar)

**Data Exploration**

 - [x] charts (Umar)
 - [x] correlations (Umar)
 - [x] descriptive_statistics (Umar)

**Data Ingestion**

 - [x] data_ingestion (Umar)

**Data Preparation**

 - [x] decomposition (Badr)
 - [x] duplicates (Badr)
 - [x] encoding (Badr)
 - [x] features_engineering (Badr)
 - [x] joins (Badr)
 - [x] missing_values (Badr)
 - [x] normalization (Badr)
 - [x] outliers (Badr)

**Machine Learning**

 - [x] classification (Badr)
 - [x] clustering (Badr)
 - [x] introduction (Badr)
 - [x] model_tracking_versioning (Umar)
 - [x] regression (Badr)
 - [x] time_series (Umar)

**Full Stack**

 - [x] complex_data_vmap
 - [x] dblink_integration
 - [x] geopandas
 - [x] interpolate
 - [x] linear_regression
 - [x] to_json
 - [x] train_test_split
 - [x] udf
 - [x] vdataframe_magic

**Performance**

 - [x] queryprofiler (Umar)

**Comment by oualib** [comment]
Part 1: https://github.com/vertica/VerticaPy/pull/1315

**Comment by oualib** [comment]
First part: https://github.com/vertica/VerticaPy/pull/1320
